src/librustc_resolve/lib.rs

```diff
@@ -1025,6 +1029,8 @@ pub struct Resolver<'a> {
     privacy_errors: Vec<PrivacyError<'a>>,
 
     arenas: &'a ResolverArenas<'a>,
+
+    is_static_method: bool,
```

Comment:
Rather than adding a new field to this struct, you could add a `is_static` boolean parameter to the `MethodRibKind` constructor of `RibKind`, whose goal is exactly of indicating wether we are currently inside of a method definition.